### PR TITLE
[CI] Ensure that VS has the correct version of Xcode in all jobs.

### PR DESCRIPTION
**Pull request by mandel-macaque**
I have pushed to origin to test in all possible cases (we have two diff pools, one for CI one for PRs).

### CODE REVIEW

**Comment by dalexsoto** [comment]
@mandel-macaque do we have these anywhere?

```
DeleteSafe (appleSdkOverride);
CreateSetting (appleSdkOverride, "AppleSdkRoot", GetSelectedXcodePath ());
```

**Review by dalexsoto — APPROVED** [review]

File: tools/devops/build-provisioning.csx.in [hunk 1]
```diff
@@ -14,3 +14,28 @@ Item ("@MONO_PACKAGE@");
...
+
+void CreateSetting (string settingFile, string key, string value)
+{
+	Exec ("defaults", "write", settingFile, key, value);
+}
+
+string GetSettingValue (string settingFile, string keyName)
+{
+	return Exec ("defaults", "read", settingFile, keyName).FirstOrDefault ();
+}
```
Comment (rolfbjarne) [inline tools/devops/build-provisioning.csx.in:41]: 👁️

File: tools/devops/device-tests-provisioning.csx.in [hunk 2]
```diff
@@ -14,6 +14,32 @@ Item ("@MONO_PACKAGE@");
...
+
+void CreateSetting (string settingFile, string key, string value)
+{
+	Exec ("defaults", "write", settingFile, key, value);
+}
+
+string GetSettingValue (string settingFile, string keyName)
+{
+	return Exec ("defaults", "read", settingFile, keyName).FirstOrDefault ();
+}
```
Comment (rolfbjarne) [inline tools/devops/device-tests-provisioning.csx.in:45]: 👀

File: tools/devops/mac-tests-provisioning.csx.in [hunk 3]
```diff
@@ -13,3 +13,28 @@ Item ("@XM_PACKAGE@");
...
+
+void CreateSetting (string settingFile, string key, string value)
+{
+	Exec ("defaults", "write", settingFile, key, value);
+}
+
+string GetSettingValue (string settingFile, string keyName)
+{
+	return Exec ("defaults", "read", settingFile, keyName).FirstOrDefault ();
+}
```
Comment (rolfbjarne) [inline tools/devops/mac-tests-provisioning.csx.in:40]: 👓

**Review by rolfbjarne — APPROVED** [review]

File: tools/devops/build-provisioning.csx.in (hunk 1 above)
Comment (mandel-macaque) [inline tools/devops/build-provisioning.csx.in:41]:
```suggestion
}

```

File: tools/devops/device-tests-provisioning.csx.in (hunk 2 above)
Comment (mandel-macaque) [inline tools/devops/device-tests-provisioning.csx.in:45]:
```suggestion
}

```

File: tools/devops/mac-tests-provisioning.csx.in (hunk 3 above)
Comment (mandel-macaque) [inline tools/devops/mac-tests-provisioning.csx.in:40]:
```suggestion
}

```

**Pull request merged**